src/components/inputs/date-time-input/date-time-input.spec.js

```diff
@@ -1,33 +1,263 @@
+    });
+    // 12 pm (12:00)
+    expect(parseTime('12 pm')).toEqual({
+      hours: 12,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    // 11 pm (23:00)
+    expect(parseTime('11 pm')).toEqual({
+      hours: 23,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    // 12:30 pm (12:30)
+    expect(parseTime('12:30 pm')).toEqual({
+      hours: 12,
+      minutes: 30,
+      seconds: 0,
+      milliseconds: 0,
+    });
+  });
+
+  it('should work with 24h times', () => {
+    expect(parseTime('0:00')).toEqual({
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    expect(parseTime('12:00')).toEqual({
+      hours: 12,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    expect(parseTime('23:59')).toEqual({
+      hours: 23,
+      minutes: 59,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    expect(parseTime('23:59:34.010')).toEqual({
+      hours: 23,
+      minutes: 59,
+      seconds: 34,
+      milliseconds: 10,
+    });
+    expect(parseTime('23:59:34.001')).toEqual({
+      hours: 23,
+      minutes: 59,
+      seconds: 34,
+      milliseconds: 1,
+    });
+    expect(parseTime('23:59:34.1')).toEqual({
+      hours: 23,
+      minutes: 59,
+      seconds: 34,
+      milliseconds: 100,
+    });
+  });
+
+  it('should work with some invalid 24:00 time for convenience', () => {
+    expect(parseTime('24:00')).toEqual({
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+    expect(parseTime('24:30')).toEqual(null);
+  });
 });
 
-describe('<DateTimeInput />', () => {
-  beforeEach(() => {
-    isTouchDevice.mockClear();
-    isTouchDevice.mockReturnValue(false);
+const baseProps = { value: '', onChange: () => {}, timeZone: 'UTC' };
+
+class TestComponent extends React.Component {
+  state = {
+    value: '',
+  };
+  handleChange = value => {
+    this.setState({ value });
+  };
+  render() {
+    return (
+      <div>
+        <label htmlFor="some-id">Date</label>
+        <DateTimeInput
+          {...baseProps}
+          id="some-id"
+          name="some-name"
+          value={this.state.value}
+          onChange={this.handleChange}
+        />
+        <div data-testid="value">{this.state.value}</div>
+      </div>
+    );
+  }
+}
+
+describe('DateTimeInput', () => {
+  it('should forward data-attributes', () => {
+    const { container } = render(
+      <DateTimeInput {...baseProps} data-foo="bar" />
+    );
+    expect(container.querySelector('[data-foo="bar"]')).toBeTruthy();
+  });
+
+  it('should have an HTML name', () => {
+    const { container } = render(<DateTimeInput {...baseProps} name="foo" />);
+    expect(container.querySelector('[name="foo"]')).toBeTruthy();
   });
-  it('should render', () => {
-    expect(shallow(<DateTimeInput {...createTestProps()} />)).toMatchSnapshot();
+
+  it('should allow changing the value by typing in english', () => {
```

Comment:
`typing in english`?

Reply:
isn't the idea that you type in German format and you get US format back? Maybe the test could be renamed to reflect this? BTW there's nothing English about US date formatting 😠

Reply:
Yeah, true. I'll change that 👍 

"typing in english" was meant to indicate "typing a date in US format"